lib/ansible/plugins/inventory/vultr.py

```diff
@@ -0,0 +1,99 @@
+    description:
+        - Get inventory hosts from Vultr public cloud.
+        - Uses ~/.vultr.ini, ./vultr.ini or VULTR_API_CONFIG path to config file.
+    options:
+        plugin:
+            description: token that ensures this is a source file for the 'vultr' plugin.
+            required: True
+            choices: ['vultr']
+        api_account:
+            description: Specify the account to be used.
+'''
+
+EXAMPLES = r'''
+# vultr_inventory.yml file in YAML format
+# Example command line: ansible-inventory --list -i vultr_inventory.yml
+
+plugin: vultr
+'''
+
+import json
+import os
+
+from ansible.errors import AnsibleError
+from ansible.plugins.inventory import BaseInventoryPlugin
+from ansible.module_utils.six.moves import configparser
+from ansible.module_utils.urls import open_url
+from ansible.module_utils._text import to_native
+
+
+def _retrieve_api_key(api_account):
+    if os.environ.get('VULTR_API_KEY'):
+        return os.environ.get('VULTR_API_KEY')
+    else:
+        paths = []
+        if os.environ.get('VULTR_API_CONFIG'):
+            paths += (os.path.expanduser(os.environ.get('VULTR_API_CONFIG'),))
+        paths += (
+            os.path.join(os.path.expanduser('~'), '.vultr.ini'),
+            os.path.join(os.getcwd(), 'vultr.ini'),
+        )
+
+        conf = configparser.ConfigParser()
+        conf.read(paths)
+
+        if not conf._sections.get(api_account):
+            return None
+
+        return dict(conf.items(api_account)).get('key')
+
+
+def _retrieve_servers(api_key):
+    api_url = 'https://api.vultr.com/v1/server/list'
+
+    try:
+        response = open_url(
+            api_url, headers={'API-Key': api_key, 'Content-type': 'application/json'}
+        )
+        return json.loads(response.read())
+    except ValueError as e:
+        raise AnsibleError("Incorrect JSON payload")
+    except Exception as e:
+        raise AnsibleError("Error while fetching %s: %s" % (api_url, to_native(e)))
+
+
+def _normalize_servers_list(servers_list):
+    return [server for id, server in servers_list.items()]
+
+
+class InventoryModule(BaseInventoryPlugin):
+    NAME = 'vultr'
+
+    def parse(self, inventory, loader, path, cache=True):
+        super(InventoryModule, self).parse(inventory, loader, path)
+        self._read_config_data(path=path)
+        api_account = self.get_option('api_account') or 'default'
+        api_key = _retrieve_api_key(api_account)
+
+        servers = _normalize_servers_list(_retrieve_servers(api_key))
+        for server in servers:
+            for group in ['location', 'os']:
+                self.inventory.add_group(group=server[group])
+                self.inventory.add_host(group=server[group], host=server['label'])
+
+            for attribute, value in server.items():
+                self.inventory.set_variable(server['label'], attribute, value)
+            self.inventory.set_variable(server['label'], 'ansible_host', server['main_ip'])
```

Comment:
User of this inventory plugin should be able to use `v6_main_ip` for value of `ansible_host` (i don't know if it makes sens to be able to use `internal_ip`) or no `ansible_host` at all (meaning that `server['label']` will be used instead).

Reply:
Done. (But unsure if this is what was expected - let me know if it needs change)